crates/typst-syntax/src/lexer.rs

```diff
@@ -353,6 +350,19 @@ impl Lexer<'_> {
         self.s.jump(end);
     }
 
+    fn inline_raw(&mut self, start: usize, end: usize, backticks: usize) {
+        self.s.jump(start + backticks);
+        let inner = self.s.to(end - backticks);
+        for line in split_newlines(inner).into_iter() {
+            self.s.eat_newline();
+            self.push_raw(SyntaxKind::RawTrimmed);
+            self.s.advance(line.len());
+            self.push_raw(SyntaxKind::Text);
+        }
```

Comment:
I think lexing directly with the scanner rather than relying on `split_newlines` would be pretty simple and more efficient here.